File: src/main/java/ru/ioleksiv/telegram/bot/api/annotations/filter/telegram/PassportDataFilter.java
package ru.ioleksiv.telegram.bot.api.annotations.filter.telegram;

import ru.ioleksiv.telegram.bot.api.model.annotation.AnnotationState;
import ru.ioleksiv.telegram.bot.api.model.annotation.CustomValidator;
import ru.ioleksiv.telegram.bot.api.model.annotation.stub.StubPassportDataValidator;
import ru.ioleksiv.telegram.bot.api.model.objects.passport.PassportData;

import java.lang.annotation.Documented;
import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

@Retention(RetentionPolicy.RUNTIME)
@Target(ElementType.METHOD)
@Documented
public @interface PassportDataFilter {

    AnnotationState status() default AnnotationState.ON;

    Class<? extends CustomValidator<PassportData>>[] validator() default StubPassportDataValidator.class;

    EncryptedPassportElementArrayFilter encryptedPassportElements() default @EncryptedPassportElementArrayFilter(status = AnnotationState.OFF);

    EncryptedCredentialsFilter encryptedCredentials() default @EncryptedCredentialsFilter(status = AnnotationState.OFF);

}

File: core/src/main/kotlin/io/github/oleksivio/tl/kbot/core/annotations/filter/composite/PassportDataFilter.kt
package io.github.oleksivio.tl.kbot.core.annotations.filter.composite

import io.github.oleksivio.tl.kbot.core.model.annotation.AnnotationState

@Retention
@Target(AnnotationTarget.FUNCTION, AnnotationTarget.PROPERTY_GETTER, AnnotationTarget.PROPERTY_SETTER)
@MustBeDocumented
annotation class PassportDataFilter(
    val status: AnnotationState = AnnotationState.ON,
    val validator: Array<String> = [],
    val encryptedPassportElements: EncryptedPassportElementArrayFilter = EncryptedPassportElementArrayFilter(
        status = AnnotationState.OFF
    ),
    val encryptedCredentials: EncryptedCredentialsFilter = EncryptedCredentialsFilter(
        status = AnnotationState.OFF
    )
)
